pynamodb/asyncio/connection.py

```diff
@@ -97,30 +94,28 @@
 
 
 class AsyncPynamoDBContext(AbstractAsyncContextManager):
-    def __init__(self):
-        current_ctx = ClientContext.get()
-        if current_ctx is not None:
-            raise RuntimeError(
-                "You're already inside an async context. Only a single `AsyncPynamoDB` context is allowed."
-            )
+    CLEANUP_TIMEOUT = 10
 
-        stack = contextlib.AsyncExitStack()
-        self._token = ClientContext.set(stack)
+    def __init__(self):
+        self._context_token = None
+        self._client_token = None
 
-    def __aenter__(self):
-        stack = ClientContext.get()
-        if not stack:
-            raise RuntimeError(
-                "You're not inside an async context. You must use `AsyncPynamoDB` to create a connection."
-            )
-        return stack.__aenter__()
+    async def __aenter__(self):
+        stack, token = await create_stack()
+        self._client_token = await create_client_stack()
+        self._context_token = token
+        return await stack.__aenter__()
 
     async def __aexit__(self, exc_type, exc_val, exc_tb):
-        stack = ClientContext.get()
-        if not stack:
-            return
-        await stack.__aexit__(exc_type, exc_val, exc_tb)
-        ClientContext.reset(self._token)
+        stack = get_stack()
+        with anyio.fail_after(self.CLEANUP_TIMEOUT, shield=True):
```

Comment:
Would it make sense to have a possibility for a longer timeout passed into the constructor, for example for backfill queries which may run a long time?

Reply:
Will include that